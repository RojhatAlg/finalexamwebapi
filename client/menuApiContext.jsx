import React from "react";
import { fetchJSON } from "./fetchJSON.jsx";

export const MenuApiContext = React.createContext({
    async listMenu(dishname){
        return await fetchJSON(`/api/movies?titleSearch=${dishname}`);
    },
    async createMenu(menu){
      fetch("/api/menu", {
          method: "post",
          body: JSON.stringify(menu),
          headers: {
            "content-type": "application/json",
          },
      });
    },
});