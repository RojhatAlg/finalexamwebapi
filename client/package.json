{
  "name": "client",
  "version": "1.0.0",
  "description": "",
  "type": "module",
  "scripts": {
    "build": "npm install && parcel build index.html",
    "dev": "parcel watch index.html",
    "test": "jest",
    "verify": "jest --collectCoverage"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@babel/preset-env": "^7.20.2",
    "@babel/preset-react": "^7.18.6",
    "babel-jest": "^29.2.2",
    "jest": "^29.2.2",
    "jest-environment-jsdom": "^29.2.2",
    "jsdom": "^20.0.1",
    "parcel": "^2.7.0",
    "process": "^0.11.10"
  },
  "dependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-icons": "^4.6.0",
    "react-router-dom": "^6.4.2"
  },
  "babel": {
    "presets": [
      "@babel/preset-env",
      "@babel/preset-react"
    ]
  },
  "jest": {
    "roots": ["<rootDir>/src"],
    "testEnvironment": "jsdom",
    "globals": {
      "IS_REACT_ACT_ENVIRONMENT": true,
      "transform": {
        "\\.js$": "<rootDir>/node_modules/babel-jest"
      }
    },
    "collectCoverageFrom": [
      "**/*.{jsx,js}",
      "!index.jsx",
      "!coverage/**",
      "!dist/**"
    ]
  },
  "development": [
    "last 1 chrome version",
    "last 1 firefox version",
    "last 1 safari version"
  ]
}
