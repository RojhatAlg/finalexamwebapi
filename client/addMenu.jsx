import * as React from "react";
import {Link, useNavigate} from "react-router-dom";
import { useContext, useState } from "react";
import { MenuApiContext } from "./menuApiContext.jsx";
import {submitMenuMethode} from "./fetchApi/api.js";



export function AddMenu(){
    const navigate = useNavigate();
    const [dishname, setDishname] = useState("");
    const [category, setCategory] = useState("");
    const [allergy, setAllergy] = useState("");
    const [error, setError] = useState(false)

    const onchangDishname=(e)=> {
        const result = e.target.value;
        if(result ==null){
            setError(false)
        }
        setDishname(result)


    }
    const onchangCategory=(e)=> {
        const result = e.target.value;
        if(result ==null){
            setError(false)
        }
        setCategory(result)

    }

    const onchangAllergy=(e)=> {
        const result = e.target.value;
        if(result ==null){
            setError(false)
        }
        setAllergy(result)

    }

    const onSubmit=(e) =>{
        e.preventDefault()
        const obj ={
            dishname,
            category,
            allergy
        }
        console.log(obj)
        submitMenuMethode(obj)
        navigate('/')



    }



    return(
        <div>
            <div className="nav">
                <ul>
                    <li className={"nav-detail"}>   <Link to='/'>Homepage</Link>
                    </li>
                </ul>
            </div>
            {error && <div className="h1">plear fot fill the input</div>}
            <form className="form-group" onSubmit={onSubmit}>
                <h2>register the form</h2>
                <label>Dish name: </label>
                <input name='dishname' placeholder="dishname" type="text" value={dishname} onChange={onchangDishname} />
                <label>Category: </label>
                <input name='category' placeholder="category" type="text" value={category} onChange={onchangCategory} />
                <label>Allergy: </label>
                <input name='allergy' placeholder="allergy" type="text" value={allergy} onChange={onchangAllergy}/>
                <button type='submit'>register</button>
            </form>
        </div>

    );
}

