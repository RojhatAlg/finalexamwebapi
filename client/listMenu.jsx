import React, {useContext, useEffect, useState} from "react";
import {MenuApiContext} from "./menuApiContext.jsx";
import {Link} from "react-router-dom";


export function ListMenu() {



    const [data, setData] = useState([]);

    const fetchData = () => {
        fetch(`/api/menus`)
            .then((response) => response.json())
            .then((actualData) => {
                console.log(actualData);
                setData(actualData);
                console.log(data);
            })
            .catch((err) => {
                console.log(err.message);
            });
    };

    useEffect(() => {
        fetchData();
    }, []);




    return (
        <div>
            <div className="nav">
                <ul>
                    <li className={"nav-detail"}>   <Link to='/'>Homepage</Link>
                    </li>
                </ul>
            </div>
            <tbody>
            <tr>
                <th>Dish Name: </th>
                <th>Category: </th>
                <th>Allergy: </th>
            </tr>
            {data.map((item) => (
                <tr>
                    <td>{item.dishname}</td>
                    <td>{item.category}</td>
                    <td>{item.allergy}</td>
                </tr>
            ))}
            </tbody>

        </div>
    );
}

export default ListMenu;