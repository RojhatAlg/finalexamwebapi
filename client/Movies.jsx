import * as React from "react";
import {BrowserRouter, Link, Route, Routes } from "react-router-dom";
import {FaSignInAlt,FaUser} from 'react-icons/fa'

import {AddMenu} from "./addMenu.jsx";
import ListMenu from "./listMenu.jsx";
import Login from "./users/login.js";
import Register from "./users/register.js";
import './App.css';
import {useEffect} from "react";


export function FrontPage() {


    return(
        <div>
            <div className="nav">
                <ul>
                    <li className={"nav-detail"}>   <Link to='/'>Homepage</Link>
                    </li>
                </ul>
                <ul>
                    <li><Link to="/login"><FaSignInAlt/>login</Link></li>
                    <li><Link to="/register"><FaUser/>register</Link></li>
                </ul>
            </div>

            <h1> Welcome to catering business </h1>
            <ul>
                <li>
                    <Link to={"/list"}> List Menu </Link>
                </li>
                <li>
                    <Link to={"/add"}> Add new Movie </Link>
                </li>
            </ul>
        </div>
    );
}

export function Movies(){
    return(
        <Routes>
            <Route path={"/list"} element={<ListMenu />} />
            <Route path={"/add"} element={<AddMenu />} />
        </Routes>
    );
}

export function Application() {
    return(
        <BrowserRouter>
            <Routes>
                <Route path={"/list"} element={<ListMenu />} />
                <Route path={"/add"} element={<AddMenu />} />
                <Route path={"/"} element={<FrontPage />} />
                <Route path='/login' element={<Login/>}/>
                <Route path='/register' element={<Register/>}/>
            </Routes>
        </BrowserRouter>
    );
}