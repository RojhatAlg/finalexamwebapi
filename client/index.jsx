import * as React from "react";
import { createRoot } from "react-dom/client";

import { Application } from "./Movies.jsx";
import ListMenu from "./listMenu.jsx";


const element = document.getElementById("app");
const root = createRoot(element);

root.render(<Application />);