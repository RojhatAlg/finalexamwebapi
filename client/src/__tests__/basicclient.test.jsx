import * as React from "react"
import {createRoot} from "react-dom/client";
import { act } from "react-dom/test-utils";
import { MenuApiContext } from "../../menuApiContext.jsx";

import { ListMenu } from "../../Movies.jsx";

const movies = [
    {
        dishname: "Dishname Example",
        category: "Category Example",
        allergy: "Allergy Example"
    }
]

async function renderListMovies(listMenus) {
    const element = document.createElement("div");
    const root = createRoot(element);

    await act(async () =>
        root.render(
            <MenuApiContext.Provider value={{ listMenus }}>
                <ListMenu />
            </MenuApiContext.Provider>
        )
    );
    return element;
}

describe("client test suite", () => {

    it("show loading screen", async () => {
        const element = await renderListMovies(() => new Promise(() => {}));

        expect(element.querySelector(".loading-indicator")).not.toBeNull();
        expect(element.innerHTML).toMatchSnapshot();
    });

    it("shows movies list", async () => {
        const element = await renderListMovies(async () => movies);

        expect(element.querySelector("h3").innerHTML).toEqual(` ${movies[0].dishname} - ${movies[0].category} - ${movies[0].allergy} `);
        expect(element.innerHTML).toMatchSnapshot();
    });

    it("shows error message", async () => {
        const element = await renderListMovies(async () => {
           throw new Error("Failed to fetch");
        });

        expect(element.querySelector(".error-message").innerHTML)
            .toContain("Failed to fetch");
        expect(element.innerHTML).toMatchSnapshot();

    });
});