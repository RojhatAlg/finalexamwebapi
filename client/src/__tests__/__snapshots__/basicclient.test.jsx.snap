// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client test suite shows movies list 1`] = `"<div>
                                                                 <div className="nav">
                                                                     <ul>
                                                                         <li className={"nav-detail"}>   <Link to='/'>Homepage</Link>
                                                                         </li>
                                                                     </ul>
                                                                     <ul>
                                                                         <li><Link to="/login"><FaSignInAlt/>login</Link></li>
                                                                         <li><Link to="/register"><FaUser/>register</Link></li>
                                                                     </ul>
                                                                 </div>

                                                                 <h1> Back to the movies </h1>
                                                                 <ul>
                                                                     <li>
                                                                         <Link to={"/list"}> List Menu </Link>
                                                                     </li>
                                                                     <li>
                                                                         <Link to={"/add"}> Add new Movie </Link>
                                                                     </li>
                                                                 </ul>
                                                             </div>"`;
